### PR TITLE
Do not enforce amp-analytics 'requests' field for FIE

### PR DESCRIPTION
**Pull request by twintwox**

### CODE REVIEW

**Comment by twintwox** [comment]
@lannka @keithwrightbos @zhouyx

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -497,7 +497,7 @@ export class AmpAnalytics extends AMP.BaseElement {
    */
   generateRequests_() {
     if (!this.config_['requests']) {
-      if (!this.isInabox_) {
+      if (!this.isInabox_ || !this.isInFie_) {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Please use the `allowParentPostMessage_` method instead.

Comment (twintwox) [reply extensions/amp-analytics/0.1/amp-analytics.js]: Done

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 2]
```diff
@@ -497,7 +497,7 @@ export class AmpAnalytics extends AMP.BaseElement {
    */
   generateRequests_() {
     if (!this.config_['requests']) {
-      if (!this.isInabox_) {
+      if (!this.isInabox_ || !this.allowParentPostMessage_) {
```
[the remark this answers is not in the document]
Comment (twintwox) [reply extensions/amp-analytics/0.1/amp-analytics.js]: Done, thanks

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 3]
```diff
@@ -497,7 +497,7 @@ export class AmpAnalytics extends AMP.BaseElement {
    */
   generateRequests_() {
     if (!this.config_['requests']) {
-      if (!this.isInabox_) {
+      if (!this.allowParentPostMessage_) {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/amp-analytics.js]: `allowedParentPostMessage_` is a method. Please use `this.allowParentPostMessage_()` Thank you.

Comment (twintwox) [reply extensions/amp-analytics/0.1/amp-analytics.js]: Done, sorry about that.

**Pull request merged**